io/net/network_test.py

```diff
@@ -62,14 +62,16 @@ def setUp(self):
         self.iface = interface
         self.ipaddr = self.params.get("host_ip", default="")
         self.netmask = self.params.get("netmask", default="")
+        self.ip_config = self.params.get("ip_config", default=True)
         local = LocalHost()
         self.networkinterface = NetworkInterface(self.iface, local)
-        try:
-            self.networkinterface.add_ipaddr(self.ipaddr, self.netmask)
-            self.networkinterface.save(self.ipaddr, self.netmask)
-        except Exception:
-            self.networkinterface.save(self.ipaddr, self.netmask)
-        self.networkinterface.bring_up()
+        if self.ip_config:
+            try:
+                self.networkinterface.add_ipaddr(self.ipaddr, self.netmask)
+                self.networkinterface.save(self.ipaddr, self.netmask)
+            except Exception:
+                self.networkinterface.save(self.ipaddr, self.netmask)
+            self.networkinterface.bring_up()
```

Comment:
@smruti77 We need to handle in `tearDown()` as well.

Reply:
@narasimhan-v ya while sending patch forget to add.